frontend/src/tests/user-permission-tests.js

```diff
@@ -14,42 +14,142 @@ import { databaseSeeder } from "./setup";
 export function userPermissionsTests(api) {
     const { apiGET, apiPOST } = api;
 
-    const newUserData = {
-        utorid: "userxx",
+    let defaultUser;
+
+    const instructorOnlyUserData = {
+        utorid: "user_permission_test_instructor_only_utorid",
         roles: ["instructor"],
     };
 
+    const TAOnlyUserData = {
+        utorid: "user_permission_test_ta_only_utorid",
+        roles: ["ta"],
+    };
+
+    const TAInstructorUserData = {
+        utorid: "user_permission_test_ta_and_instructor_utorid",
+        roles: ["ta", "instructor"],
+    };
+
+    /**
+     * Switch current active user to the user represented by the given user data. If given user does not exist, one will
+     * be created.
+     * @param newActiveUser the user to be set to the active user
+     * @returns {Promise<void>}
+     */
+    async function switchToTargetUser(newActiveUser) {
+        // Set the active user to the target user
+        let resp = await apiPOST("/debug/active_user", newActiveUser);
+        expect(resp).toHaveStatus("success");
+    }
+
+    /**
+     * Restores the active user to the default user (the user logged during test setup in beforeAll).
+     *
+     * @returns {Promise<void>}
+     */
+    async function restoreDefaultUser() {
+        let respSwitchBackUser = await apiPOST(
```

Comment:
If you're not doing something meaningful with the response objects, just name them `resp`.